.github/workflows/checks.yml

```diff
@@ -154,28 +160,19 @@ jobs:
           name: contract-artifacts
           path: starknet_py/tests/e2e/mock/
 
-      - uses: actions/setup-python@v4
-        with:
-          python-version: "3.12"
-          cache: 'pip'
-
```

Comment:
Looks like this removal may affect `pip` caching? Same remark to other places where it's removed.